ui/src/app/pages/analysis/runtime-panel.tsx

```diff
@@ -40,97 +40,171 @@ const styles = reactStyles({
-                min={50 /* Runtime API has a minimum 50GB requirement. */}/>
-  </div>;
+const DiskSizeSelection = ({onChange, updatedDiskSize, diskSize}) => {
+  return <Fragment>
+    <label htmlFor='runtime-disk' style={{marginRight: '.25rem'}}>Disk (GB)</label>
+    <InputNumber id='runtime-disk'
+      showButtons
+      decrementButtonClassName='p-button-secondary'
+      incrementButtonClassName='p-button-secondary'
+      value={updatedDiskSize || diskSize}
+      inputStyle={{padding: '.75rem .5rem', width: '2rem'}}
+      onChange={({value}) => onChange(value)}
+      min={50 /* Runtime API has a minimum 50GB requirement. */}/>
+  </Fragment>;
+};
+
+const DataProcConfig = ({onChange, dataprocConfig}) => {
+  const {
+    workerMachineType = defaultMachineType,
+    workerDiskSize = 50,
+    numberOfWorkers = 2,
+    numberOfPreemptibleWorkers = 0
+  } = dataprocConfig || {};
+  const [updatedNumWorkers, setUpdatedNumWorkers] = useState(numberOfWorkers);
+  const [updatedPreemtible, setUpdatedPreemptible] = useState(numberOfPreemptibleWorkers);
+  const [updatedWorkerMachine, setUpdatedWorkerMachine] = useState(null);
+  const [updatedDiskSize, setUpdatedDiskSize] = useState(workerDiskSize);
+
+  useEffect(() => {
+    const machineType = updatedWorkerMachine && updatedWorkerMachine.name;
+    const dataprocConfigChanged = updatedNumWorkers !== numberOfWorkers ||
+    updatedPreemtible !== numberOfPreemptibleWorkers ||
+    updatedDiskSize !== workerDiskSize ||
+    updatedWorkerMachine ||
+    !dataprocConfig;
+
+    onChange(dataprocConfigChanged ? {
+      workerMachineType: machineType,
+      workerDiskSize: updatedDiskSize,
+      numberOfWorkers: updatedNumWorkers,
+      numberOfPreemptibleWorkers: updatedPreemtible
+    } : null);
+
+    return () => onChange(null);
+  }, [updatedNumWorkers, updatedPreemtible, updatedWorkerMachine, updatedDiskSize]);
+
+
+  return <fieldset style={{marginTop: '0.75rem'}}>
+    <legend style={styles.workerConfigLabel}>Worker Config</legend>
+    <div style={styles.formGrid}>
+      <label htmlFor='num-workers' style={{marginRight: '.25rem'}}>Workers</label>
+      <InputNumber id='num-workers'
+        showButtons
+        decrementButtonClassName='p-button-secondary'
+        incrementButtonClassName='p-button-secondary'
+        value={updatedNumWorkers}
+        inputStyle={{padding: '.75rem .5rem', width: '2rem'}}
+        onChange={({value}) => {
+          setUpdatedNumWorkers(value);
+          if (updatedNumWorkers < updatedPreemtible) {
+            setUpdatedPreemptible(updatedNumWorkers);
+          }
+        }}
+        min={2}/>
+      <label htmlFor='num-preemptible' style={{marginRight: '.25rem'}}>Preemptible</label>
+      <InputNumber id='num-preemptible'
+        showButtons
+        decrementButtonClassName='p-button-secondary'
+        incrementButtonClassName='p-button-secondary'
+        value={updatedNumWorkers < updatedPreemtible ? updatedNumWorkers : updatedPreemtible}
```

Comment:
Nice. I actually did not think it worked this way (I think later we might need to slightly clarify our UX to make this clearer). I'm asking Leo team to clarify their docs on this, since I actually didn't find a definitive answer (but I assume this is correct, since Terra UI does it this way).

Reply:
Implications per slack updates: just let the preemtible worker count update independently of numWorkers.